File: app/src/main/java/com/github2136/picker/GlideLoader.java
package com.github2136.picker;

import android.content.Context;
import android.widget.ImageView;

import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.github2136.picturepicker.other.ImageLoader;

/**
 * Created by yb on 2018/5/16.
 */

public class GlideLoader implements ImageLoader {

    @Override
    public void loadThumbnail(Context context, int resizeX, int resizeY, ImageView imageView, String path) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .override(resizeX, resizeY)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .centerCrop()
                .dontAnimate()
                .into(imageView);
    }

    @Override
    public void loadAnimatedGifThumbnail(Context context, int resizeX, int resizeY, ImageView imageView, String path) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .override(resizeX, resizeY)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .centerCrop()
                .into(imageView);
    }

    @Override
    public void loadImage(Context context, ImageView imageView, String path) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL.ALL)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .dontAnimate()
                .into(imageView);
    }

    @Override
    public void loadAnimatedGifImage(Context context, ImageView imageView, String path) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL.ALL)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .into(imageView);
    }

    @Override
    public boolean supportAnimatedGifThumbnail() {
        return true;
    }

    @Override
    public boolean supportAnimatedGif() {
        return true;
    }
}

File: app/src/main/java/com/github2136/picker/GlideLoader.kt
package com.github2136.picker

import android.content.Context
import android.widget.ImageView

import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.github2136.picturepicker.other.ImageLoader

/**
 * Created by yb on 2018/5/16.
 */
class GlideLoader : ImageLoader {

    override fun loadThumbnail(context: Context, resizeX: Int, resizeY: Int, imageView: ImageView, path: String) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .override(resizeX, resizeY)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .centerCrop()
                .dontAnimate()
                .into(imageView)
    }

    override fun loadAnimatedGifThumbnail(context: Context, resizeX: Int, resizeY: Int, imageView: ImageView, path: String) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .override(resizeX, resizeY)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .centerCrop()
                .into(imageView)
    }

    override fun loadImage(context: Context, imageView: ImageView, path: String) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .dontAnimate()
                .into(imageView)
    }

    override fun loadAnimatedGifImage(context: Context, imageView: ImageView, path: String) {
        GlideApp.with(context)
                .load(path)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .placeholder(R.drawable.img_picker_place)
                .error(R.drawable.img_picker_fail)
                .into(imageView)
    }

    override fun supportAnimatedGifThumbnail(): Boolean {
        return true
    }

    override fun supportAnimatedGif(): Boolean {
        return true
    }
}
